### DISCUSSION TITLE
xml question - referencing a contained item within a contained item

### DISCUSSION THREAD

**Discussion by wolf-jonathan** [FakeFishGames/Barotrauma#8066] [Code questions and modding support]
Is there any way to apply a StatusEffect on an item that is contained within a contained item?

To give context - 
Suppose I am trying to allow the player to insert a flashlight into a revolver, like an attachment. When that happens the gun produces light, just like a flashlight. However, I am not sure if it is possible for the gun to start draining the battery inside the flashlight, as the battery is not contained in the gun, but rather in the flashlight in the gun. So we have a contained item buried inside another contained item. 
Is there currently a way in the xml to reference that?

**Comment by Terragen** [comment]
Sounds similar to how the diving suit lockers work - you put a diving suit in them and, if powered, they refill an oxygen tank contained inside that diving suit.

Looks like the StatusEffect is of type "OnContained" (when the suit is in the suit locker) and the target is "contained" (the oxygen tank inside the suit)

from divinggear.xml:

`
      

      <StatusEffect type="OnContained" target="Contained" Condition="1.0">

        <Conditional Voltage="gt 0.01" targetcontainer="true" targetgrandparent="true" targetitemcomponent="Powered" />

        <RequiredItem items="refillableoxygensource" type="Contained" excludebroken="false" />

      </StatusEffect>
`

Perhaps you can model your solution on that existing code - I'd guess you'd need to update the code for the gun to allow it to contain a battery as well as the code on the flashlight so that it drains the battery contained inside it when it is contained by a pistol (and not any other container that can hold it).

Hope that helps!
